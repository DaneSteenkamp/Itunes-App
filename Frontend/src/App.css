.jumbotron {
  text-align: center;
  background-image: linear-gradient(
    to bottom right,
    tomato,
    rgb(247, 118, 139)
  );
}

.jumboHeading {
  font-family: "SF Pro Display", "SF Pro Icons", "Helvetica Neue", "Helvetica",
    "Arial", sans-serif;
  font-size: 80px;
  color: white;
}

.jumboWriteUp {
  font-family: "SF Pro Display", "SF Pro Icons", "Helvetica Neue", "Helvetica",
    "Arial", sans-serif;
  font-size: 24px;
  color: white;
}

.dropDown {
  background-color: white;
  color: black;
  text-decoration: none;
  border: none;
  padding: 10px;
  width: 200px;
  text-align: center;
  border-radius: 50px;
  text-transform: uppercase;
}

.searchInput {
  margin-left: 35px;
}

.returnData {
  text-align: center;
}
