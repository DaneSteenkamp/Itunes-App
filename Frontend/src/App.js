import React, { Component } from "react";
import "./App.css";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.min.css";
import Jumbotron from "react-bootstrap/Jumbotron";

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      results: [],
      query: "City Color",
      entity: "musicTrack",
      favourites: [],
    };
    this.fetchItunes = this.fetchItunes.bind(this);
    this.addtoFav = this.addtoFav.bind(this);
  }

  handleChange = (name) => (e) => {
    // Captures the data entered into the input
    this.setState({
      [name]: e.target.value,
    });
  };

  fetchItunes() {
    // retrives data from Itunes databse and stores it in the empty array results
    axios
      .get(`/search/${this.state.query}/${this.state.entity}`)
      .then((responce) =>
        this.setState({
          results: responce.data,
        })
      );
  }

  componentDidMount() {
    this.fetchItunes();
  }

  addtoFav(fav) {
    // This function gets called whenever somone click the add button. The data gets stored in a new array called favourites
    this.setState({
      favourites: [...this.state.favourites, fav],
    });
  }

  removeFav(id) {
    // removes a item from the favourites array
    const updatedFav = this.state.favourites.filter((f) => f.trackId !== id);
    this.setState({
      favourites: updatedFav,
    });
  }

  render() {
    console.log(this.state.favourites);
    const categories = [
      { Music: "musicTrack" },
      { Movies: "movie" },
      { Podcast: "podcast" },
      { Ebook: "ebook" },
    ];

    return (
      <div>
        <Jumbotron>
          <div className="jumboContainer">
            <h1 className="jumboHeading">iTunes search app</h1>
            <p className="jumboWriteUp">
              Search iTunes database for your favourite Music, Movies, Books and
              Podcasts
            </p>
          </div>
          <select onChange={this.handleChange("entity")} className="dropDown">
            {categories.map((cat, index) => {
              const key = Object.keys(cat);
              const value = Object.values(cat);
              return (
                <option key={index} value={value}>
                  {key}
                </option>
              );
            })}
          </select>
          <input
            className="searchInput"
            placeholder="search"
            onChange={this.handleChange("query")}
          />
          <button onClick={this.fetchItunes}>Submit</button>
        </Jumbotron>
        <div className="container">
          <div className="row">
            {this.state.results.map((itunesData) => (
              <div className="returnData col-sm" key={itunesData.artistID}>
                <h2 style={{ color: "tomato" }}>Artist / Title:</h2>
                <h2>{itunesData.artistName}</h2>
                <h2 style={{ color: "tomato" }}>Name:</h2>
                <h2>{itunesData.trackName}</h2>
                <button onClick={() => this.addtoFav(itunesData)}>Add</button>
              </div>
            ))}
          </div>
        </div>
        <Jumbotron>
          <div className="jumboContainer">
            <h1 className="jumboHeading">Favourites</h1>
          </div>
        </Jumbotron>
        <div className="container">
          <div className="row">
            {this.state.favourites.map((itunesData) => (
              <div className="returnData col-sm" key={itunesData.artistID}>
                <h2 style={{ color: "tomato" }}>Artist / Title:</h2>
                <h2>{itunesData.artistName}</h2>
                <h2 style={{ color: "tomato" }}>Name:</h2>
                <h2>{itunesData.trackName}</h2>
                <button onClick={() => this.removeFav(itunesData.trackId)}>
                  Remove
                </button>
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

export default App;
