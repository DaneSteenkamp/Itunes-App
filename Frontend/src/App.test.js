import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

describe("snapshop testing", () => {
  test("snapshot for App Component", () => {
    // Creating a snapshot of the App component
    const renderedComponent = renderer.create(<App />).toJSON();
    expect(renderedComponent).toMatchSnapshot(); // toMatch method is used to see if the JSON representation of the DOM tree matched the snapshot
  });
});
