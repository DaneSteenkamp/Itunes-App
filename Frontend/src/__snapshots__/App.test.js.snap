// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshop testing snapshot for App Component 1`] = `
<div>
  <div
    className="jumbotron"
  >
    <div
      className="jumboContainer"
    >
      <h1
        className="jumboHeading"
      >
        iTunes search app
      </h1>
      <p
        className="jumboWriteUp"
      >
        Search iTunes database for your favourite Music, Movies, Books and Podcasts
      </p>
    </div>
    <select
      className="dropDown"
      onChange={[Function]}
    >
      <option
        value={
          Array [
            "musicTrack",
          ]
        }
      >
        Music
      </option>
      <option
        value={
          Array [
            "movie",
          ]
        }
      >
        Movies
      </option>
      <option
        value={
          Array [
            "podcast",
          ]
        }
      >
        Podcast
      </option>
      <option
        value={
          Array [
            "ebook",
          ]
        }
      >
        Ebook
      </option>
    </select>
    <input
      className="searchInput"
      onChange={[Function]}
      placeholder="search"
    />
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
  <div
    className="jumbotron"
  >
    <div
      className="jumboContainer"
    >
      <h1
        className="jumboHeading"
      >
        Favourites
      </h1>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
